src/game/client/swarm/c_asw_weapon.cpp

```diff
@@ -885,31 +914,103 @@ void C_ASW_Weapon::SimulateLaserPointer()
 	m_bLocalPlayerControlling = bLocalPlayer;
 }
 
-void C_ASW_Weapon::CreateLaserPointerEffect( bool bLocalPlayer, int iAttachment )
+int C_ASW_Weapon::GetFreeLaserIndex(bool bSetAsTaken)
 {
-	if ( m_pLaserPointerEffect )
+	int index = 0;
+	while (index < 8) //LS INDEX SIZE 8
```

Comment:
There can be up to ASW_MAX_MARINE_RESOURCES marines on the map in, for example, deathmatch.